### PR TITLE
upgrade to pug

### PR DESCRIPTION
**Pull request by lightbringer1991**
change the library to handle `.pug` files

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,111 +1,111 @@
-var path = require('path'),
```
Comment (lightbringer1991) [inline index.js]: move to ES6 syntax

File: index.js [hunk 2]
```diff
@@ -1,111 +1,111 @@
...
-  fs = require('fs'),
-  jade = require('jade'),
-  url = require('url'),
-  assert = require('assert');
-
-var defaultOptions = {
-  // Jade options
-  jade: {},
-  // Serve index.jade if http://someurl/example/ is requested
+const _ = require('lodash');
```
Comment (omgaz) [inline index.js]: Can we do away with this and:

assigns can be `{...oldProps, ...newProps}`
includes can be `[].indexOf(val) != -1`
startsWith can be `string[0] == something`

Comment (lightbringer1991) [reply index.js]: okay I can do that

File: index.js [hunk 3]
```diff
@@ -1,72 +1,79 @@
...
+
+    res.setHeader('Content-Length', Buffer.byteLength(html));
+    res.setHeader('Content-Type', 'text/html; charset=utf-8');
+    const now = +new Date();
+    const expires = new Date(now + newOpts.maxAge);
+    res.setHeader('Cache-Control', `max-age=${newOpts.maxAge}`);
+    res.setHeader('Expires', expires.toGMTString());
+    return res.end(html);
+  });
+  });
```
Comment (Maxmus1989) [inline index.js]: indentation?

File: index.js [hunk 4]
```diff
@@ -1,72 +1,79 @@
...
+  allowedExt: ['.pug', '.htm', '.html'],
   // Header for Cache-Control: max-age=0
-  maxAge: 0
+  maxAge: 0,
 };
 
-module.exports = function(opts) {
 
-  if (!opts.baseDir)
+module.exports = function (opts) {
```
Comment (Maxmus1989) [inline index.js]: arrow function?

**Comment by vinteo** [comment]
THere are a few other libraries like https://github.com/ryan-schroeder/pug-static and https://github.com/Nattfarinn/connect-pug-static, do they not serve our purpose?

File: index.js [hunk 5]
```diff
@@ -1,72 +1,79 @@
...
+    }
 
+    pug.renderFile(filePath, newOpts.pug, (renderErr, html) => {
+      if (renderErr) {
+        return next(renderErr);
+      }
+
+    res.setHeader('Content-Length', Buffer.byteLength(html));
+    res.setHeader('Content-Type', 'text/html; charset=utf-8');
+    const now = +new Date();
```
Comment (Maxmus1989) [inline index.js]: might be confusing as `now` in timestamp while `expires` as Date. maybe rename variable?

File: package.json [hunk 6]
```diff
@@ -1,18 +1,18 @@
 {
-  "name": "connect-jade-static",
+  "name": "connect-pug-static",
   "version": "0.2.2",
```
Comment (vinteo) [inline package.json]: 0.1.0?

**Comment by omgaz** [comment]
lgtm just sort out version

**Comment by vinteo** [comment]
maybe rename to `express-pug-static` so we can publish it @bchew

**Comment by lightbringer1991** [comment]
@vinteo `connect-pug-static` doesn't really work. `pug-static` doesn't separate source and destination path, so it doesn't really work in our case without changing the code.

**Comment by vinteo** [comment]
Merging...

**Pull request merged**